File: src/SignalRChat.Server/wwwroot/js/avatarService.js
/**
 * AvatarService — generates avatar URLs for users.
 * Mirrors the server-side IAvatarService for client-side previews.
 * SRP: avatar logic only.
 */

const BASE_URL = 'https://api.dicebear.com/7.x/bottts-neutral/svg';
const COLORS = '5865f2,57f287,fee75c,eb459e,ed4245';

export function getAvatarUrl(userName) {
    if (!userName) return `${BASE_URL}?seed=anonymous&backgroundColor=${COLORS}`;
    const seed = encodeURIComponent(userName.trim().toLowerCase());
    return `${BASE_URL}?seed=${seed}&backgroundColor=${COLORS}`;
}

File: src/SignalRChat.Server/Services/AvatarService.cs
using SignalRChat.Server.Interfaces;

namespace SignalRChat.Server.Services;

/// <summary>
/// Generates deterministic avatar URLs using DiceBear's free API.
/// Same username = same avatar = persistent identity.
/// </summary>
public class AvatarService : IAvatarService
{
    private const string BaseUrl = "https://api.dicebear.com/7.x/bottts-neutral/svg";

    public string GetAvatarUrl(string userName)
    {
        var seed = Uri.EscapeDataString(userName.Trim().ToLowerInvariant());
        return $"{BaseUrl}?seed={seed}&backgroundColor=5865f2,57f287,fee75c,eb459e,ed4245";
    }
}
